src/structured_config.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Structured configuration for the Kafka charm."""
+import logging
+from enum import Enum
+from typing import Optional
+
+from charms.data_platform_libs.v0.data_models import BaseConfigModel
+from pydantic import validator
+
+logger = logging.getLogger(__name__)
+
+
+class LogMessageTimestampType(str, Enum):
+    """Enum for the `log_message_timestamp_type` field."""
+
+    CREATE_TIME = "CreateTime"
+    LOG_APPEND_TIME = "LogAppendTime"
+
+
+class LogCleanupPolicy(str, Enum):
+    """Enum for the `log_cleanup_policy` field."""
+
+    COMPACT = "compact"
+    DELETE = "delete"
+
+
+class CompressionType(str, Enum):
+    """Enum for the `compression_type` field."""
+
+    GZIP = "gzip"
+    SNAPPY = "snappy"
+    LZ4 = "lz4"
+    ZSTD = "zstd"
+    UNCOMPRESSED = "uncompressed"
+    PRODUCER = "producer"
+
+
+class CharmConfig(BaseConfigModel):
+    """Manager for the structured configuration."""
+
+    compression_type: str
```

Comment:
**question** But the type here...shouldn't this be CompressionType? And similar for other enum types?